### ISSUE TITLE
Allow hidden version check HTTP call to be disabled

### ISSUE THREAD

**Issue by NJAldwin** [spotify/scio#2169]
Each time a `ScioContext` class is initialized, scio makes a HTTP call to GitHub to check if there's a newer version.  If there is, it logs a warning.

The warning itself can be disabled with the system property `scio.ignoreVersionWarning`.  However, this just disables the warning messages -- the HTTP call is still made!

This should absolutely be configurable (ideally from code rather than solely from a global system property).  Anything that goes wrong with an external HTTP call like this could potentially put our production systems at risk, and it's completely unnecessary for teams that are prudent about keeping track of versions.

This should all be documented, as well.  Especially since it currently defaults to always making this call, but there's no indication of it anywhere in the docs or in the logs unless there's a new version available.

In summary:
 - ScioContext makes a hidden HTTP call every time a new one is instantiated
 - This cannot currently be disabled
 - This should be able to be disabled, ideally via a code param rather than a system property
 - The fact that the check is made, as well as the ways to disable the check, should be documented.

**Comment by NJAldwin** [comment]
Thanks for tweaking the flag!  However, I don't really consider this fully closed.  You still have dangerous behavior that's enabled by default and yet is completely undocumented.

**Comment by NJAldwin** [comment]
#2175 ^
